### PR TITLE
wear backpack in ventral position

### PR DESCRIPTION
**Pull request by MarcBook**
Thanks @commy2

### CODE REVIEW

**Comment by jonpas** [comment]
Can you change it to comply with coding guidelines, specifically newlines and indentation, please.

File: addons/backpacks/functions/fnc_backpackToBack.sqf [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ */
+
+#include "script_component.hpp"
+
+_unit = _this select 0;
+
+hint format ["%1", attachedObjects _unit];
+
+_backpack = (attachedObjects _unit) select 0;
```
Comment (PabstMirror) [inline addons/backpacks/functions/fnc_backpackToBack.sqf]: If user has lights attached, the backpack might not be the first in attachedObjects

**Comment by MarcBook** [comment]
Ready for mergin

File: addons/backpacks/functions/fnc_playerVehicleChanged.sqf [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#include "script_component.hpp"
+
+private ["_unit", "_holder"];
+
+_unit = _this select 0;
+_holder = _unit getVariable "ACE_Backpack" select 0;
```
Comment (PabstMirror) [inline addons/backpacks/functions/fnc_playerVehicleChanged.sqf]: No guaranteed that this variable is set.  _holder will be nil
probably want `getVariable ["ACE_Backpack", [objNull, false]]`

**Comment by KoffeinFlummi** [comment]
Please fix the indentation to use 4 spaces in all spaces (including the stringtable; we still need to do that for most of ours).

**Comment by PabstMirror** [comment]
The variable `ace_backpack` doesn't get reset during respawn.

I had a backpack attached to front, died, and when respawning I had an option to move to front.
It threw:
```
19:20:09 Error in expression <lect _i;
detach _backpack;
};
};

_unit action ["AddBag",_backpack,(backpackCarg>
19:20:09   Error position: <action ["AddBag",_backpack,(backpackCarg>
19:20:09   Error Type Any, expected String
19:20:09 File z\ace\addons\backpacks\functions\fnc_backpackToBack.sqf, line 24
```

File: addons/backpacks/functions/fnc_playerVehicleChanged.sqf [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+if (vehicle _unit != _unit) then {
+	if (vehicle _unit isKindOf "ParachuteBase") then {
+		_holder attachTo [vehicle _unit, [-0.12,0.65,-0.15]];
+		_holder setVectorDirAndUp [[0,-0.2,-1],[0,1,0]];
+	} else {
+		_holder setPos [0, 0, 200];
```
Comment (commy2) [inline addons/backpacks/functions/fnc_playerVehicleChanged.sqf]: ? Whats the point of that?

File: addons/backpacks/functions/fnc_backpackToBack.sqf [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+for "_i" from 0 to (count attachedObjects _unit - 1) step 1 do {
+	if (attachedObjects _unit select _i isKindOf "groundWeaponHolder") then {
+		_backpack = attachedObjects _unit select _i;
+		detach _backpack;
+	};
+};
+
+_unit action ["AddBag",_backpack,(backpackCargo _backpack) select 0];
+_unit setVariable ["ACE_Backpack", [objNull, false]];
```
Comment (Glowbal) [inline addons/backpacks/functions/fnc_backpackToBack.sqf]: * Why use an array for this? You switch between `[objNull, false]` and `[backpack, true]`. You could use just `backpack` and check if the value `isNull`.
* You should be able to ditch the for loop is way, and just check the `ACE_Backpack` variable instead.
* Missing line end.
* Please complete the function header with `Return value:`.

**Comment by MarcBook** [comment]
Now it should work.
Tested in MP

File: addons/backpacks/CfgVehicles.hpp [hunk 5]
```diff
@@ -0,0 +1,29 @@
+class CfgVehicles {
+    class Man;
+    class CAManBase: Man {
+        class ACE_SelfActions {
+            class ACE_Equipment {
+                class ACE_BackpackToFront {
+                    displayName = "$STR_ACE_Backpack_BackpackToFront";
+                    condition = "(backpack _player != '') and (vehicle _player == _player) and (isNull (_player getVariable ['ACE_Backpack', objNull]))";
```
Comment (commy2) [inline addons/backpacks/CfgVehicles.hpp]: vehicle _player == _player

can be removed. checked as trivial condition anyway

exception would be "isNotInside"

File: addons/backpacks/functions/fnc_playerVehicleChanged.sqf [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+_unit = _this select 0;
+_holder = _unit getVariable ["ACE_Backpack", objNull];
+
+if (vehicle _unit != _unit) then {
+    if (vehicle _unit isKindOf "ParachuteBase") then {
+        _holder attachTo [vehicle _unit, [-0.12,0.65,-0.15]];
+        _holder setVectorDirAndUp [[0,-0.2,-1],[0,1,0]];
+    } else {
+        _holder setPos [0, 0, 200];
+	};
```
Comment (commy2) [inline addons/backpacks/functions/fnc_playerVehicleChanged.sqf]: Tab, unmergable ;-)

File: addons/backpacks/CfgVehicles.hpp [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+                    exceptions[] = {"isNotInside"};
+                    statement = QUOTE([_player] call FUNC(backpackToFront));
+                    showDisabled = 0;
+                    priority = 2.5;
+                    icon = "\A3\ui_f\data\map\VehicleIcons\iconbackpack_ca.paa";
+                    hotkey = "B";
+                };
+                class ACE_BackpackToBack {
+                    displayName = "$STR_ACE_Backpack_BackpackToBack";
+                    condition = "(backpack _player == '') and (vehicle _player == _player) and (!isNull (_player getVariable ['ACE_Backpack', objNull]))";
```
Comment (commy2) [inline addons/backpacks/CfgVehicles.hpp]: same

File: addons/backpacks/functions/fnc_playerVehicleChanged.sqf (hunk 6 above)
Comment (bux578) [inline addons/backpacks/functions/fnc_playerVehicleChanged.sqf]: :heart: for the w40k reference

**Comment by alganthe** [comment]
does not seems to work as of 2015-04-26 tried like this:
* placed a player unit 
* placed a Huron flying at 1000m
* made the player unit in cargo inside the Huron
* used self interaction to strap the backpack to my hairy chest
* grabbed a parachute and jumped off
* after landing I noticed the backpack was not there.

**Comment by commy2** [comment]
I can reproduce the missing backpack issue (while trying to put it on the back).
I'll look into it.

File: addons/backpacks/CfgVehicles.hpp [hunk 8]
```diff
@@ -0,0 +1,29 @@
+class CfgVehicles {
+    class Man;
+    class CAManBase: Man {
+        class ACE_SelfActions {
+            class ACE_Equipment {
+                class ACE_BackpackToFront {
+                    displayName = "$STR_ACE_Backpack_BackpackToFront";
+                    condition = "(backpack _player != '') and (isNull (_player getVariable ['ACE_Backpack', objNull]))";
+                    exceptions[] = {"isNotInside"};
```
Comment (MarcBook) [inline addons/backpacks/CfgVehicles.hpp]: Normally you should not be able to do this.

**Comment by commy2** [comment]
@MarcBook I'm still working on this. There are some issues with simply attaching a GWH to a unit, as these things tend to suck in other items that are dropped nearby. Also other units can take simply take the backpack which leeds to other glitches.
Don't worry about fixing those. I'll take over.

**Comment by MarcBook** [comment]
@commy2 
Thanks :)

**Comment by ViperMaul** [comment]
Status check?

**Comment by esteldunedain** [comment]
@ViperMaul, this won't happen any time soon, due to backpacks looking super weird

**Comment by WastedMike** [comment]
This is a must have, and I have faint in the ACE team.

**Comment by sargken** [comment]
Status on what needs to be done?

**Comment by bux578** [comment]
Sorry @MarcBook you're working speed is just freaking too slow

**Comment by RyanMcc101** [comment]
From what I understand this pull request is closed due to major bugs. Just wanted to check if this mod had managed to work around the bugs or has the same issues you ran into

http://www.armaholic.com/page.php?id=28837

**Comment by alganthe** [comment]
Current engine limitations makes placing any type of container on your chest very wonky.
Check #2794 for more informations.

**Comment by bux578** [comment]
>  this pull request is closed 

This PR is closed because it hasn't been worked on since April last year.

**Pull request closed without merging**